addons/l10n_es_edi_verifactu_pos/models/pos_order.py

```diff
@@ -0,0 +1,127 @@
+from odoo import _, api, models
+from odoo.exceptions import UserError, ValidationError
+
+
+class PosOrder(models.Model):
+    _name = 'pos.order'
+    _inherit = ['pos.order', 'l10n_es_edi_verifactu.record_mixin']
+
+    @api.depends('country_code')
+    def _compute_l10n_es_edi_verifactu_required(self):
+        # Overrides verifactu_record_mixin.py
+        for order in self:
+            order.l10n_es_edi_verifactu_required = order.country_code == 'ES' and order.company_id.l10n_es_edi_verifactu_required
+
+    @api.depends('company_id', 'name', 'date_order', 'amount_total')
+    def _compute_l10n_es_edi_verifactu_record_identifier(self):
+        for order in self:
+            if not order.l10n_es_edi_verifactu_required:
+                identifier = {
+                    'errors': [_("Veri*Factu is not enabled for the point of sale order.")],
+                }
+            elif order.state == 'draft':
+                identifier = {
+                    'errors': [_("The point of sale order is in draft.")],
+                }
+            else:
+                identifier = self.env['l10n_es_edi_verifactu.document']._record_identifier(
+                    order.company_id, order.name, order.date_order, order.amount_total
+                )
+            order.l10n_es_edi_verifactu_record_identifier = identifier
```

Comment:
Another option is keeping the fields readonly?  Because e.g. what happens if you cancel, but one of the elements was changed?  (nif e.g.)